utils/bitarray_utils.py

```diff
@@ -18,7 +26,7 @@ def uint_to_bitarray(x: int, bit_width=None) -> BitArray:
     converts an unsigned into to bits.
     if bit_width is provided then data is converted accordingly
     """
-    return int2ba(x, length=bit_width)
+    return int2ba(int(x), length=bit_width)
```

Comment:
why converting to int? Is it better to give an error than to silently convert to int

Reply:
yeah, not needed it looks like

Reply:
added an assert